Client/mods/deathmatch/logic/CClientVehicle.cpp

```diff
@@ -5023,3 +5024,35 @@ void CClientVehicle::ResetWheelScale()
 
     m_bWheelScaleChanged = false;
 }
+
+CVector CClientVehicle::GetEntryPoint(int entryPointIndex)
+{
+    const auto GetSADoorIndex = [&entryPointIndex]()
+    {
+        switch (entryPointIndex)
+        {
+            case 0:
+                return 10;
+            case 1:
+                return 8;
+            case 2:
+                return 11;
+            case 3:
+                return 9;
+        }
+
+        return 0;
+    };
+
+    unsigned int saDoorIndex = GetSADoorIndex();
```

Comment:
Why bother creating a lambda? Just assign the value directly

Reply:
Would be going in circles here. Earlier review from Pirulax in 2022 made me change from directly assigning to using the lambda.

Reply:
> Would be going in circles here. Earlier review from Pirulax in 2022 made me change from directly assigning to using the lambda.

How so? This would be exactly the same thing
```cpp
std::uint32_t saDoorIndex = 0;

switch (entryPointIndex) {
    case 0:
        saDoorIndex = 10;
        break;
    case 1:
        saDoorIndex = 8;
        break;
    case 2:
        saDoorIndex = 11;
        break;
    case 3:
        saDoorIndex = 9;
        break;
}
```

Reply:
I agree and will change it back. Just hope nobody comes around to make me change it again lol. This is how I originally had it.

Reply:
I prefer using arrays:

```
uint32_t DOOR_LOOOKUP[4] = {
 10, 8, 11, 9
};

uint32_t saDoorIndex = DOOR_LOOOKUP[entryPointIndex];
```

Reply:
This could be moved out of the function and marked constexpr then.

Reply:
I decided to stick to your first suggestion Tracer. Perfectly readable and acceptable.